LearningMissionSimulation/PlaygroundGavril.cs

```diff
@@ -237,7 +262,39 @@ public void CreateClassrooms(int classroomCount)
 
         public void AssignInstructorsToClassrooms()
         {
-            throw new NotImplementedException();
+
+            string action = "Assign Instructors To Classrooms";
+            ReportHeader(actionName: action);
+
+            if (classroomList.Count == 0)
+            {
+                ReportError(missingResource: "Classroom", failedAction: action);
+            }
+            else
+            {
+                foreach (var classroom in classroomList)
+                {
+                    List<Instructor> instructorList;
+                    if (ModuleInstructorDictionary.TryGetValue(classroom.Module.Id, out instructorList))
+                    {
+                        if (instructorList.Count > 0)
+                        {
+                          var i =  classroom.Head = instructorList[AttributeGenerator.random.Next(0, instructorList.Count)];
```

Comment:
This is very wrong! I don't even understand what you are trying to achieve here. Please remove "var i =".  I asked you to count a classroom when you add a head(instructor) to it. it is a very easy task. You have to maintain and increment an index every time you assign a head to a classroom Please fix it.